python/gop_routes.py

```
@@ -0,0 +1,235 @@
+    if not dsn:
+        return []
+    with _ro_lock:
+        try:
+            import psycopg
+
+            if _ro_conn is None or _ro_conn.closed:
+                _ro_conn = psycopg.connect(dsn, connect_timeout=5, autocommit=True)
+            with _ro_conn.cursor() as cur:
+                cur.execute(sql, params or {})
+                cols = [d.name for d in cur.description]
+                return [dict(zip(cols, r)) for r in cur.fetchall()]
```

Ruff (0.15.20)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)